benchmarks/gc_ratchet/gc_ratchet.py

```
@@ -533,6 +581,165 @@ def measure(
+        "host": host_description(),
+        "run_config": {"repeats": repeats, "warmup": warmup, "scan_mode_env": SCAN_MODE_ENV},
+        "probes": rows,
+    }
+
+
+def render_classification(payload: Mapping[str, Any]) -> str:
+    lines = [
+        "## GC ratchet retention classification",
+        "",
+        "`conservative` is what `heap_used_bytes` gates on: measured after the probe's",
+        f"own `gc()`, which forces the conservative native-stack scan. `precise` is the",
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)